skore/src/skore/sklearn/_comparison/metrics_accessor.py

```diff
@@ -157,14 +164,175 @@ def report_metrics(
+                new_index = ["Metric", "Label / Average", "Estimator"]
+            else:
+                new_index = ["Metric", "Estimator"]
+            df = df.reset_index().set_index(new_index)
+
+            # Then drop the model from the columns
+            df.columns = df.columns.droplevel(0)
+
+            return df
+
+        def melt_df(df):
+            df_reset = df.reset_index()
+
+            if "Label / Average" in df_reset.columns:
+                id_vars = ["Metric", "Label / Average", "Estimator"]
+            else:
+                id_vars = ["Metric", "Estimator"]
+
+            melted = pd.melt(
+                df_reset,
+                id_vars=id_vars,
+                var_name="Split",
+                value_name="Value",
+            )
+
+            return melted
+
+        results = individual_results.copy()
+
+        # Pop the favorability column if it exists, to:
+        # - not use it in the aggregate operation
+        # - later to only report a single column and not by split columns
+        if indicator_favorability:
+            favorability = results[0]["Favorability"]
+            for result in results:
+                result.pop("Favorability")
+        else:
+            favorability = None
+
+        dfs_model_name_in_index = [
+            add_model_name_to_index(df, estimator_name)
+            for df, estimator_name in zip(results, estimator_names)
+        ]
+
+        dfs_melted = [melt_df(df) for df in dfs_model_name_in_index]
+
+        df = pd.concat(dfs_melted, axis=0)
+
+        if aggregate:
+            if isinstance(aggregate, str):
+                aggregate = [aggregate]
+
+            if "Label / Average" in df.columns:
+                index = ["Metric", "Label / Average"]
+            else:
+                index = ["Metric"]
+
+            df = df.pivot_table(
+                index=index,
+                columns="Estimator",
+                values="Value",
+                aggfunc=aggregate,
+                sort=False,
+            )
+        else:
+            # Make sure the Split column goes [0, 1, 2, 0, 1, 2]
+            # Rather than [0, 0, 1, 1, 2, 2]
+
+            metric_order = df["Metric"].unique()
+
+            df["metric_order_index"] = df["Metric"].apply(
+                lambda x: list(metric_order).index(x)
+            )
+
+            if "Label / Average" in df.columns:
+                by = ["metric_order_index", "Label / Average", "Estimator", "Split"]
+            else:
+                by = ["metric_order_index", "Estimator", "Split"]
+
+            df = (
+                df.sort_values(by=by)
+                .drop("metric_order_index", axis=1)
+                .reset_index(drop=True)
+            )
+
+            new_index = []
+            for col in df.columns:
+                if col == "Value":
+                    break
+                new_index.append(col)
+            df = df.set_index(new_index)
```

Comment:
Isn't it equivalent to:

```suggestion
            df = df.set_index(df.columns.drop("Value"))
```

Reply:
Can you provide me a minimum example where "Value" is not the last column?

Reply:
In case that we really have this case, we can do:

```python
df = df.set_index(df.columns[:df.get_loc["Value"]])
```

Reply:
> Can you provide me a minimum example where "Value" is not the last column?

No, all tests pass. I applied your first suggestion